fabfile.py

```diff
@@ -844,6 +848,111 @@ def updatemaincode(backup=False):
 
             venvcmd_py3('pip install -r ../url_metadata/requirements.txt')
 
+        if env.uses_bluenove_actionable:
+            execute(update_bluenove_actionable)
+
+
+def get_robot_machine():
+    """
+    Return the configured robot machine: (the first configured machine)
+    """
+    # Retrive the list of registered machines
+    # Machines format: machine_id,machine_name,machine_password/...others
+    machines = env.get('machines', '')
+    if machines:
+        # Get the first machine
+        robot = machines.split('/')[0]
+        # Retrive the machine data
+        robot_data = robot.split(',')
+        # We must find three data (identifier, name and password)
+        if len(robot_data) != 3:
+            print red("The data of the user machine are wrong! %s" % robot)
+            return None
+
+        return {
+            'identifier': robot_data[0].strip(),
+            'name': robot_data[1].strip(),
+            'password': robot_data[2].strip()
+        }
+
+    print red("No user machine found!")
+    return None
+
+
+@task
+def install_bluenove_actionable():
+    """
+    Install the bluenove_actionable app.
+    """
+    if not exists("%(projectpath)s/../bluenove-actionable/" % env):
+        print cyan("Cloning git bluenove-actionable repository")
+        with cd("%(projectpath)s/.." % env):
+            # We need an ssh access
+            run('git clone git://github.com/bluenove/bluenove-actionable.git')
```

Comment:
This is an anonymous url, you will never access the private repository with this. You need the ssh version `git@github.com:bluenove/bluenove-actionable.git`

Reply:
effectivement